back/engines/commercial/verification/app/controllers/verification/patches/omniauth_callback_controller.rb

```diff
@@ -43,11 +44,22 @@ def verification_callback(verification_method)
 
       def fail_verification(error)
         omniauth_params = request.env['omniauth.params'].except('token', 'pathname')
-        url = add_uri_params(
-          Frontend::UrlService.new.verification_failure_url(pathname: request.env['omniauth.params']['pathname']),
-          omniauth_params.merge(verification_error: true, error: error)
-        )
-        redirect_to url
+
+        # It seems the FE is not parsing verification failure URLs, so we need to use the SSO failure URL structure
+        if error == 'not_entitled_under_15_years_of_age'
+          redirect_params = {
+            sso_pathname: request.env['omniauth.params']['pathname'],
+            sso_response: true,
+            error_code: 'not_entitled_under_15_years_of_age'
+          }
+          redirect_to(add_uri_params(Frontend::UrlService.new.signin_failure_url, redirect_params))
+        else # TODO: Check if this is ever used by the FE, or anywhere else.
+          url = add_uri_params(
+            Frontend::UrlService.new.verification_failure_url(pathname: request.env['omniauth.params']['pathname']),
+            omniauth_params.merge(verification_error: true, error: error)
```

Comment:
This would give a redirect to something like `.../some-page?random-passthrough-param=somevalue&verification_error=true&error=not_entitled_under_15_years_of_age'`, given a request of ```
GET "/auth/nemlog_in?token=#{token}&random-passthrough-param=somevalue&pathname=/some-page"
      follow_redirect!
```
but it seems the FE no longer parses these URLs, which surpises me somewhat. See [Slack thread](https://citizenlabco.slack.com/archives/CQAGMHKT3/p1694605712950699).

Reply:
I get that it might not be practical from the front-end perspective to systematically check for the presence of some query parameters (verification_error, error) for all routes. However, I'd prefer to either create a counterpart to "/authentication-error" for verification errors or redirect all verification errors to «  /authentication-error» (although this second approach is still questionable).

Reply:
Reverted to using verification specific redirect URL format(s)